examples/hal/quad/main.rs

```diff
@@ -499,62 +349,186 @@ fn main() {
+                PipelineStage::TRANSFER..PipelineStage::FRAGMENT_SHADER,
                 m::Dependencies::empty(),
                 &[image_barrier],
             );
 
             cmd_buffer.finish()
         };
 
-        let submission = Submission::new()
-            .submit(Some(submit));
-        queue.submit(submission, Some(&mut frame_fence));
+        let submission = Submission::new().submit(Some(submit));
+        queue_group.queues[0].submit(submission, Some(&mut frame_fence));
 
         device.wait_for_fence(&frame_fence, !0);
     }
 
+    #[cfg(feature = "vulkan")]
+    let (
+        mut pixel_width,
+        mut pixel_height,
+        mut swap_chain,
+        mut render_pass,
+        mut framebuffers,
+        mut frame_images,
+        mut pipeline,
+        mut pipeline_layout,
+    ) = swapchain_stuff(
+        pixel_width,
+        pixel_height,
+        &mut surface,
+        &surface_format,
+        &physical_device,
+        &mut device,
+        &set_layout,
+    );
+
+    #[cfg(not(feature = "vulkan"))]
+    let (
+        mut pixel_width,
+        mut pixel_height,
+        mut swap_chain,
+        render_pass,
+        framebuffers,
+        frame_images,
+        pipeline,
+        pipeline_layout,
+    ) = swapchain_stuff(
+        pixel_width,
+        pixel_height,
+        &mut surface,
+        &surface_format,
+        &physical_device,
+        &mut device,
+        &set_layout,
+    );
+
     //
     let mut running = true;
+    let mut recreate_swapchain = false;
+    let mut first = true;
     while running {
         events_loop.poll_events(|event| {
             if let winit::Event::WindowEvent { event, .. } = event {
                 match event {
                     winit::WindowEvent::KeyboardInput {
-                        input: winit::KeyboardInput {
-                            virtual_keycode: Some(winit::VirtualKeyCode::Escape),
-                            .. },
+                        input:
+                            winit::KeyboardInput {
+                                virtual_keycode: Some(winit::VirtualKeyCode::Escape),
+                                ..
+                            },
                         ..
-                    } | winit::WindowEvent::CloseRequested => running = false,
+                    }
+                    | winit::WindowEvent::CloseRequested => running = false,
+                    winit::WindowEvent::Resized(width, height) => {
+                        pixel_width = width as _;
+                        pixel_height = height as _;
+                        if !first {
+                            recreate_swapchain = true;
+                        }
+                    }
                     _ => (),
                 }
             }
         });
+        first = false;
+
+        if recreate_swapchain {
+            device.wait_idle().unwrap();
+            unsafe { ManuallyDrop::drop(&mut viewport) }
```

Comment:
I think we need to destroy the old viewport (because it depends on the swapchain) before making  a new one. Non-vulkan backends of course don't need this, but I made it manuallydrop anyways to avoid having two separate let statements.

Reply:
but viewport is just plain data, it doesn't interact with the swapchain in any hidden way. There should be no need to hack around `drop` for the plain data